src/HaveIBeenPwned.PwnedPasswords.Downloader/Program.cs

```diff
@@ -94,13 +95,56 @@ internal sealed class PwnedPasswordsDownloader : Command<PwnedPasswordsDownloade
     private readonly HttpClient _httpClient;
     private readonly ResiliencePipeline<HttpResponseMessage> _pipeline;
 
+    private string[] _etag = new string[1024 * 1024];
+
+    private void ReadEtagFile(string filePath)
+    {
+        try
+        {
+            if (File.Exists(filePath))
+            {
+                string json;
+                using (StreamReader reader = File.OpenText(filePath))
+                {
+                    json = reader.ReadToEnd();
+                }
+                string[]? data = JsonSerializer.Deserialize<string[]>(json);
+                if (data != null)
+                {
+                    _etag = data;
+                }
+            }
+        }
+        catch (Exception ex)
+        {
+            Console.WriteLine($"Error deserializing the object: {ex.Message}");
+        }
+    }
+
+    private void WriteEtagFile(string filePath)
+    {
+        try
+        {
+            string json = JsonSerializer.Serialize(_etag);
```

Comment:
Given the content, why not serialize like this, to make the output cleaner?

```csharp
string json = JsonSerializer.Serialize(_etag, s_jsonOptions);
```

Using this:
```csharp
private static JsonSerializerOptions s_jsonOptions = new() { Encoder = System.Text.Encodings.Web.JavaScriptEncoder.UnsafeRelaxedJsonEscaping };
```